app/controllers/routing/index/CountryOfRoutingController.scala

```diff
@@ -72,24 +79,52 @@ class CountryOfRoutingController @Inject() (
             .bindFromRequest()
             .fold(
               formWithErrors => Future.successful(BadRequest(view(formWithErrors, lrn, countryList.values, mode, index))),
-              value =>
+              selectedCountry =>
                 for {
                   ctcCountries <- countriesService.getCountryCodesCTC().map(_.values)
-                  isInCL112 = ctcCountries.map(_.code.code).contains(value.code.code)
+                  isInCL112 = ctcCountries.map(_.code.code).contains(selectedCountry.code.code)
                   customsSecurityAgreementAreaCountries <- countriesService.getCustomsSecurityAgreementAreaCountries().map(_.values)
-                  isInCL147 = customsSecurityAgreementAreaCountries.map(_.code.code).contains(value.code.code)
+                  isInCL147 = customsSecurityAgreementAreaCountries.map(_.code.code).contains(selectedCountry.code.code)
+                  removeOfficesFromUserAnswers <- removeOfficesOfTransitCountries(index, request, selectedCountry)
+
                   result <- {
                     implicit val navigator: UserAnswersNavigator = navigatorProvider(mode, index)
                     CountryOfRoutingPage(index)
-                      .writeToUserAnswers(value)
+                      .writeToUserAnswers(selectedCountry)
                       .appendValue(CountryOfRoutingInCL112Page(index), isInCL112)
                       .appendValue(CountryOfRoutingInCL147Page(index), isInCL147)
                       .updateTask()
-                      .writeToSession()
+                      .writeToSessionWithUserAnswers(removeOfficesFromUserAnswers)
                       .navigate()
+
                   }
                 } yield result
             )
       }
   }
+
+  private def removeOfficesOfTransitCountries(index: Index, request: DataRequest[AnyContent], selectedCountry: Country): Future[UserAnswers] =
+    request.userAnswers.get(CountryOfRoutingPage(index)) match {
+      case Some(previousSelectedCountry) if previousSelectedCountry != selectedCountry =>
+        Future
+          .fromTry(
+            findAndRemoveOffices(request.userAnswers, OfficesOfTransitSection, OfficeOfTransitSection, OfficeOfTransitPage, previousSelectedCountry.code.code)
+          )
+      case _ => Future.successful(request.userAnswers)
+    }
+
+  private def findAndRemoveOffices(
+    userAnswers: UserAnswers,
+    array: Section[JsArray],
+    obj: Index => Section[JsObject],
+    page: Index => QuestionPage[CustomsOffice],
+    countryCode: String
+  ): Try[UserAnswers] =
+    (0 until userAnswers.get(array).length).foldRight(Try(userAnswers)) {
+      case (index, acc) =>
+        userAnswers.get(page(Index(index))) match {
+          case Some(value) if value.countryId == countryCode => acc.flatMap(_.remove(obj(Index(index))))
+          case _                                             => acc
+        }
+    }
```

Comment:
Can this not just live inside CountryOfRoutingPage cleanup?

Reply:
We initially tried to put it in there, but we needed the previous country (to remove transitOffices related to that country) but that` value: Option[Country]` was the new selected country. Maybe there was a way to pass that in?